File: Program's_Contributed_By_Contributors/C++ Programs/Time_Conversion.cpp
// C++ program to convert 12 hour to 24 hour format
#include<iostream>
using namespace std;

void convertto24(string str)
{
	int h1 = (int)str[1] - '0';
	int h2 = (int)str[0] - '0';
	int hh = (h2 * 10 + h1 % 10);

	// If time is in "AM"
	if (str[8] == 'A')
	{
		if (hh == 12)
		{
			cout << "00";
			for (int i=2; i <= 7; i++)
				cout << str[i];
		}
		else
		{
			for (int i=0; i <= 7; i++)
				cout << str[i];
		}
	}

	// If time is in "PM"
	else
	{
		if (hh == 12)
		{
			cout << "12";
			for (int i=2; i <= 7; i++)
				cout << str[i];
		}
		else
		{
			hh = hh + 12;
			cout << hh;
			for (int i=2; i <= 7; i++)
				cout << str[i];
		}
	}
}

int main()
{
string str = "09:25:30PM";
  convertto24(str);
return 0;
}

File: Program's_Contributed_By_Contributors/Python_Programs/time-conversion.py
import math
import random

# program to convert time to 24hour format string


def timeConversion(s):
    if 'PM' in s:
        hour = s[:2]
        if hour == '12':
            return s[:-2]
        else:
            return str(int(hour) + 12) + s[2:-2]
    else:
        hour = s[:2]
        if hour == '12':
            return '00' + s[2:-2]
        else:
            return s[:-2]


if __name__ == '__main__':
    s = input()
    result = timeConversion(s)
    print(result)
